include/libtorrent/copy_ptr.hpp

```diff
@@ -33,46 +33,34 @@ POSSIBILITY OF SUCH DAMAGE.
 #ifndef TORRENT_COPY_PTR
 #define TORRENT_COPY_PTR
 
+#include <memory>
+
 namespace libtorrent {
 
 	template <class T>
 	struct copy_ptr
 	{
-		copy_ptr(): m_ptr(nullptr) {}
+		copy_ptr() = default;
 		explicit copy_ptr(T* t): m_ptr(t) {}
 		copy_ptr(copy_ptr const& p): m_ptr(p.m_ptr ? new T(*p.m_ptr) : nullptr) {}
-		copy_ptr(copy_ptr&& p) noexcept : m_ptr(p.m_ptr) { p.m_ptr = nullptr; }
+		copy_ptr(copy_ptr&& p) noexcept = default;
 
-		void reset(T* t = nullptr) { delete m_ptr; m_ptr = t; }
+		void reset(T* t = nullptr) { m_ptr.reset(t); }
 		copy_ptr& operator=(copy_ptr const& p)
 		{
 			if (m_ptr == p.m_ptr) return *this;
-			T* c = p.m_ptr ? new T(*p.m_ptr) : nullptr;
-			delete m_ptr;
-			m_ptr = c;
-			return *this;
-		}
-		copy_ptr& operator=(copy_ptr&& p) noexcept {
-			if(m_ptr == p.m_ptr) return *this;
-			delete m_ptr;
-			m_ptr = p.m_ptr;
-			p.m_ptr = nullptr;
```

Comment:
this line: is it not needed?

Reply:
`p.m_ptr = nullptr`

Reply:
since ``std::unique_ptr`` already is movable, the default move assignment operator is fine.